src/clj/rems/experimental/load_simulator.clj

```diff
@@ -0,0 +1,172 @@
+            [rems.logging :refer [with-mdc]]
+            [rems.scheduler :as scheduler]
+            [rems.service.test-data :as test-data]
+            [rems.test-browser :as b]
+            [rems.experimental.simulator-util :as simu]
+            [rems.util :refer [rand-nth*]])
+  (:import [java.util.concurrent Executors ExecutorService TimeUnit]))
+
+(def ^:private task-counter (atom 0))
+(def ^:private current-tasks (atom {}))
+(def ^:private task-statistics (atom {:completed []
+                                      :failed 0}))
+
+(defn submit-new-application []
+  (simu/with-test-browser
+    (b/login-as (btu/context-get :user-id))
+    (btu/context-assoc! :cat-item (simu/get-random-catalogue-item))
+    (b/go-to-catalogue)
+    (b/add-to-cart (btu/context-get :cat-item))
+    (b/click-cart-apply)
+    (btu/context-assoc! :application-id (parse-int (b/get-application-id)))
+    (simu/fill-application-fields (btu/context-get :application-id))
+    (Thread/sleep 2000) ; wait for ui to catch up
+    (when (btu/exists? :accept-licenses-button)
+      (b/accept-licenses))
+    (b/send-application)
+    (b/logout)))
+
+;; XXX: add more actions, e.g. view pdf, return/approve/reject
+(defn handle-application []
+  (simu/with-test-browser
+    (b/login-as (btu/context-get :user-id))
+    (btu/context-assoc! :application-id (-> (btu/context-get :user-id)
+                                            (simu/get-random-todo-application)
+                                            :application/id))
+    (b/go-to-application (btu/context-get :application-id))
+    (btu/scroll-and-click :remark-action-button)
+    (let [selector {:id (b/get-field-id "Add remark")}]
+      (btu/fill-human selector (test-data/random-long-string 5)))
+    (btu/scroll-and-click :remark)
+    (btu/wait-visible :status-success)
+    (b/logout)))
+
+(defn view-application []
+  (simu/with-test-browser
+    (b/login-as (btu/context-get :user-id))
+    (b/go-to-applications)
+    (-> (btu/context-get :user-id)
+        (simu/get-random-application)
+        :application/id
+        (b/go-to-application))
+    (b/logout)))
+
+(comment
+  ; convenience for development
+  (btu/init-driver! :chrome "http://localhost:3000/" :development)
+
+  (btu/context-assoc! :user-id "alice")
+  (view-application)
+
+  (btu/context-assoc! :user-id "handler")
+  (handle-application)
+
+  (btu/context-assoc! :user-id "alice")
+  (with-redefs [simu/get-random-catalogue-item (constantly "Default workflow 2")]
+    (submit-new-application)))
+
+(defn get-task-action [user-id]
+  (let [roles (rems.db.applications/get-all-application-roles user-id)
+        actions (case (rand-nth* roles)
+                  :handler [handle-application]
+                  [submit-new-application view-application])]
+    (rand-nth* actions)))
+
+(defn get-task-user! [task-id]
+  (locking current-tasks
+    (let [current-users (map :user-id (vals @current-tasks))
+          available-users (-> (simu/get-all-users)
```

Comment:
On my machine it tries to use perf tester users which are not present on the login screen. I think overall we must still figure out a solution for specifying the users.